paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -182,19 +183,28 @@ def get_http_utilization_for_all_tasks(marathon_service_config, marathon_tasks,
 
     endpoint = endpoint.lstrip('/')
     utilization = []
-    for task in marathon_tasks:
-        try:
-            utilization.append(json_mapper(get_json_body_from_service(task.host, task.ports[0], endpoint)))
-        except requests.exceptions.Timeout:
+
+    monkey.patch_socket()
+    gevent_pool = pool.Pool(20)
+    jobs = [gevent_pool.spawn(json_mapper, get_json_body_from_service(task.host, task.ports[0], endpoint))
+            for task in marathon_tasks]
+    gevent.joinall(jobs)
+
+    for job in jobs:
+        if job.value is not None:
+            utilization.append(job.value)
+        elif isinstance(job.exception, requests.exceptions.Timeout):
             # If we time out querying an endpoint, assume the task is fully loaded
             # This won't trigger in the event of DNS error or when a request is refused
             # a requests.exception.ConnectionError is raised in those cases
             utilization.append(1.0)
             log.debug("Recieved a timeout when querying %s on %s:%s. Assuming the service is at full utilization." % (
                 marathon_service_config.get_service(), task.host, task.ports[0]))
-        except Exception as e:
-            log.debug("Caught exception when querying %s on %s:%s : %s" % (
-                marathon_service_config.get_service(), task.host, task.ports[0], str(e)))
+        elif job.exception is not None:
+            log.debug("Caught exception when querying %s on %s:%s : %s" % (marathon_service_config.get_service(),
```

Comment:
I'd probably up this from debug, it is a real error even if it's not that critical.